cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,238 @@
+			RococoRelaySender::get(),
+		);
+		//before reap reserved balance should be equal to total deposit
+		assert_eq!(reserved_balance, total_deposit);
+		assert_ok!(<RococoRelay as RococoRelayPallet>::IdentityMigrator::reap_identity(
+			rococo_runtime::RuntimeOrigin::root(),
+			RococoRelaySender::get(),
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Unreserved { who, amount }) => {
+					who: *who == RococoRelaySender::get(),
+					amount: *amount == total_deposit,
+				},
+			]
+		);
+		assert!(<RococoRelay as RococoRelayPallet>::Identity::identity(&RococoRelaySender::get())
+			.is_none());
+		let tuple_subs =
+			<RococoRelay as RococoRelayPallet>::Identity::subs_of(&RococoRelaySender::get());
+		assert_eq!(tuple_subs.1.len(), 0);
+
+		let reserved_balance = <RococoRelay as RococoRelayPallet>::Balances::reserved_balance(
+			RococoRelaySender::get(),
+		);
+		// after reap reserved balance should be 0
+		assert_eq!(reserved_balance, 0);
+		let free_bal_after_reap =
+			<RococoRelay as RococoRelayPallet>::Balances::free_balance(RococoRelaySender::get());
+
+		// free balance should be greater than before reap
+		assert!(free_bal_after_reap > free_bal_before_reap);
+	});
+
+	// 6. assert on Parachain
+	PeopleRococo::execute_with(|| {
+		type RuntimeEvent = <PeopleRococo as Chain>::RuntimeEvent;
+		let reserved_bal = <PeopleRococo as PeopleRococoPallet>::Balances::reserved_balance(
+			PeopleRococoSender::get(),
+		);
+		let id_deposit = id_deposit_parachain(&identity_parachain);
+		let subs_deposit = SubAccountDepositParachain::get();
+		let total_deposit = subs_deposit + id_deposit;
+
+		assert_expected_events!(
+			PeopleRococo,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Deposit { ..}) => {},
+				RuntimeEvent::Balances(pallet_balances::Event::Endowed {  ..}) => {},
+				RuntimeEvent::Balances(pallet_balances::Event::Reserved { who, amount }) => {
+					who: *who == PeopleRococoSender::get(),
+					amount: *amount == id_deposit,
+				},
+				RuntimeEvent::Balances(pallet_balances::Event::Reserved { who, amount }) => {
+					who: *who == PeopleRococoSender::get(),
+					amount: *amount == subs_deposit,
+				},
+				RuntimeEvent::IdentityMigrator(
+					polkadot_runtime_common::identity_migrator::Event::DepositUpdated {
+						who, identity, subs
+					}) => {
+					who: *who == PeopleRococoSender::get(),
+					identity: *identity == id_deposit,
+					subs: *subs == subs_deposit,
+				},
+				RuntimeEvent::MessageQueue(pallet_message_queue::Event::Processed { ..}) => {},
+			]
+		);
+
+		// reserved balance should be equal to total deposit calculated on the Parachain
+		assert_eq!(reserved_bal, total_deposit);
+
+		let free_bal =
+			<PeopleRococo as PeopleRococoPallet>::Balances::free_balance(PeopleRococoSender::get());
+
+		// Atleast a single Existential Deposit should be free
+		assert!(free_bal >= PEOPLE_ROCOCO_ED);
```

Comment:
In _most_ cases there should be exactly an ED remaining, but there are some cases it may vary due to the fact that we're changing `IdentityInfo` to remove `additional` and add `github` and `discord` fields. The test case should cover all combinations (but let's get this one right first). I'm hoping that the fee on the Relay Chain overestimates since the encoded length of `(key, value)` will be greater than `value`. But the test needs to make sure :). If it does not, then we need to inspect `additional` in `OnReapIdentity` to calculate the parachain side exactly.

Reply:
Ok, looking into it.

Reply:
So, I managed to account for it like so 
```
// set to * 2 in caclulate_remote_deposit runtime/rococo/src/impls.rs#L55
let para_existential_deposit = PEOPLE_ROCOCO_ED * 2;
let expected = para_existential_deposit - removed_fields_cost / 100 / 2;

assert_eq!(expected, PeopleRococoBalances::free_balance(PeopleRococoSender::get()));
```
I did this by eyeballing https://github.com/paritytech/polkadot-sdk/blob/ec3a61ed86305e5b70ce6e466cf127addf560186/polkadot/runtime/rococo/src/impls.rs#L55

The only operation I can't see justification for is the division by 2. But I noticed this was the case when looking at the diff in the assertion. Let me know. 

And yes @joepetrowski perhaps after this is merged I can write more cases checking different instances of `IdentityInfo`

Reply:
I think you're going about it the wrong way. This calculation doesn't really make sense to me.

>  write more cases checking different instances of IdentityInfo

IMO this is the right way to go about it. The goal is not to prove that we can calculate the expected deposit in the test; it is to ensure that no matter what `IdentityInfo` someone has on the Relay Chain, the handler will teleport enough of it to the parachain. It will need to have an ED left over for the reservation to be successful. So IMO it's OK for the account to have slightly more than ED on the parachain, but it cannot have less.

Reply:
Ok, I'll adjust the test.

Reply:
@joepetrowski I've added `fn identities() -> Vec<Identity>` and four test cases where we assert on each element of the Identity Vec.